pkg/api/graph/types.go

```diff
@@ -124,13 +206,17 @@ func (n ImageStreamTagNode) Object() interface{} {
 }
 
 func (n ImageStreamTagNode) String() string {
-	return fmt.Sprintf("<ImageStream %s/%s:%s>", n.Namespace, n.Name, n.Tag)
+	return fmt.Sprintf("<imagestream %s/%s:%s>", n.Namespace, n.Name, n.Tag)
 }
 
 func (*ImageStreamTagNode) Kind() int {
 	return ImageStreamTagGraphKind
 }
 
+func DockerImageRepositoryNodeName(o image.DockerImageReference) UniqueName {
+	return UniqueName(fmt.Sprintf("%d|%s/%s/%s:%s", DockerRepositoryGraphKind, o.Registry, o.Namespace, o.Name, o.Tag))
```

Comment:
You'll probably want to call `o.String()` (see https://github.com/openshift/origin/blob/master/pkg/image/api/helper.go#L128), as `o.Tag` can be empty while `o.ID` is populated.